File: coco/src/main/java/coco/ep/m1/Sub.java
package coco.ep.m1;

import coco.ep.m0.Exp;

public interface Sub<FT> extends Exp<FT>, Factory<FT> {

    Exp<FT> getLeft();

    Exp<FT> getRight();

    default Double eval() {
        return convert(getLeft()).eval() - convert(getRight()).eval();
    }
}

File: coco_csharp/CoCo/EP/M1/Sub.cs
namespace EP.M1 {
    public interface Sub<FT> : EP.M0.Exp<FT> {

        EP.Exp<FT> getLeft();

        EP.Exp<FT> getRight();

        double EP.M0.Exp<FT>.eval() {
            return convert(getLeft()).eval() - convert(getRight()).eval();
        }
    }
}